headers/xdp/xsk.h

```diff
@@ -249,6 +249,43 @@ struct xsk_socket_config {
 	__u16 bind_flags;
 };
 
+/* 
+ * The following fields should not be NULL at the same time:
+ *
+ * @rx, @tx
+ *  At least one traffic direction should be assigned for an XSk.
+ * 
+ * The following fields are optional:
+ * 
+ * @fill, @comp, @rx_size, @tx_size, @libbpf_flags, @libxdp_flags,
+ * @xdp_flags, @bind_flags
+ *  If a socket with exclusive ownership of a umem are going to be
+ *  created, keep @fill and @comp unset. If the umem is to be shared
+ *  with other sockets, set @fill and @comp to the corresponding
+ *  fields of the umem.
+ *  If the remaining fields are unset, they will be set to 
+ *  default value (see `xsk_set_xdp_socket_config()`).
+ * 
+ * Except for the fields mentioned above, none field can be set.
+ */
+struct xsk_socket_opts {
+	size_t sz;
+	struct xsk_ring_cons *rx;
+	struct xsk_ring_prod *tx;
+	struct xsk_ring_prod *fill;
+	struct xsk_ring_cons *comp;
+	__u32 rx_size;
+	__u32 tx_size;
+	union {
+		__u32 libbpf_flags;
```

Comment:
For this new interface, let us delete libbpf_flags and be done with the union. libbpf_flags was there just for compatibility with libbpf version prior to 1.0. For this new interface, we can skip this.